### DISCUSSION TITLE
why does https://nicegui.io/ refresh frequently

### DISCUSSION THREAD

**Discussion by retsyo** [zauberzeug/nicegui#2072] [Q&A]
### Question

I found that no matter https://nicegui.io/ or https://nicegui.io/documentation often refreshes or shows 
```
connection lost.
Trying to reconnect...
```

What causes such annoying behavior? Thanks.

**Comment by rodja** [comment]
It should be much better since yesterday when we released 1.4.3. Do you have a stable internet connection? The website has a 10 second reconnection timeout before it reloads. Maybe you could share the js debug log when a reload happens?

**Reply by retsyo** [reply]
I tested with MS edge. No one complaint the internet connection is not stable. However`https://nicegui.io/` says `connection lost. Trying to reconnect...` almost every 3 minutes. 

> Maybe you could share the js debug log when a reload happens?
how to do that?

**Comment by Shahin-rmz** [comment]
Actually I have same problem with my apps that make with the help of Nicegui. Why that happens and what can I do?

**Reply by rodja** [reply]
Do you run a single worker and single instance? Or have you multiple processes of your app like our NiceGUI website?

**Reply by Shahin-rmz** [reply]
I have written an app, and dockerised it.
and then run the app with `docker compose`.
I think there will be multiple instances of it, but not sure.
How can I check if it is single worker or not?

**Reply by rodja** [reply]
If you are using the normal `ui.run` it will be a single worker. Only if you dive deep and use gunicorn or similar there is a good chance to have multiple workers (which will break things, see https://github.com/zauberzeug/nicegui/discussions/1539). Let's assume that your deployment is just a single instance. If you experience page reloads in such a scenario the cause is almost certain either a very bad internet connection or [to much work on the main thread](https://github.com/zauberzeug/nicegui/wiki/FAQs#how-to-not-get-the-reconnect-popup-or-avoid-laggy-interaction).

**Reply by Shahin-rmz** [reply]
> Let's assume that your deployment is just a single instance.

yes, upon clearing up, now I know the deployment is a single instance.

> very bad internet connection

No, because the apps that are deployed on localhost, also refresh frequently. 

> [to much work on the main thread](https://github.com/zauberzeug/nicegui/wiki/FAQs#how-to-not-get-the-reconnect-popup-or-avoid-laggy-interaction).


It is fully correct.
So that is also the reason why when I close the page and open it, the last data that already has been inserted to database are still on fields!?

Actually I did not know what is the problem, and searched for it and did not get the answer.
So should make a page with `async` decorator.
(Will post the the question on another thread)
